File: swift/880_DecodedStringAtIndex.swift
class Solution {
    func decodeAtIndex(_ S: String, _ K: Int) -> String {
        var k = K
        var size = 0
        for c in S {
            if c.isNumber {
                size *= Int(String(c))!
            } else {
                size += 1
            }
        }
        
        for (i, c) in S.enumerated().reversed() {
            k %= size
            if c.isNumber {
                size /= Int(String(c))!
            } else {
                if k == 0 {
                    return String(c)
                }
                size -= 1
            }
        }
        return String(S[S.startIndex])
    }
}

File: python/880_DecodedStringAtIndex.py
class Solution:
    def decodeAtIndex(self, S: str, K: int) -> str:
        size = 0
        for c in S:
            if c.isnumeric():
                size *= int(c)
            else:
                size += 1
        
        for i in range(len(S) - 1, -1, -1):
            K %= size
            if S[i].isnumeric():
                size /= int(S[i])
            else:
                if K == 0: return S[i]
                size -= 1
        return S[0]
